core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/StackSensitiveDataPropagator.java

```diff
@@ -16,8 +18,9 @@
     @Inject
     private CredentialSourceDecorator credentialSourceDecorator;
 
-    public Stack propagate(CredentialSourceRequest credentialSourceRequest, Stack stack, Workspace workspace) {
-        if (credentialSourceRequest != null) {
+    public Stack propagate(StackRequest stackRequest, Stack stack, Workspace workspace) {
+        CredentialSourceRequest credentialSourceRequest = stackRequest.getCredentialSource();
+        if (StringUtils.isEmpty(stackRequest.getEnvironment()) && credentialSourceRequest != null) {
```

Comment:
Why is the additional condition needed here? Decorator does not use the environment if I am not terribly mistaken.

Reply:
This code / class will be removed during API clarification (it is from the HDC era), but till that if environment is given for a stack, then the environment's credential will be used for the stack.